### DISCUSSION TITLE
Nonnegative weights

### DISCUSSION THREAD

**Discussion by Rasmuskh** [google/flax#2481] [General] [answered]
Hi, 
What is a good way to initialize weights to be nonnegative in flax?

My current approach is to rewrite the glorot_uniform and variance_scaling functions, to do what I want. That works but also adds a lot of boilerplate code to my project.

I hope there is a nicer/cleaner way to do this.

**Comment by cgarciae** [comment, marked as answer]
@Rasmuskh Creating custom initializers is fine, if they are taking too much space consider creating an `initializers.py` module and then import them from there.

**Reply by Rasmuskh** [reply]
Thanks,
Yes, moving all the custom initializers to a separate module would make things cleaner.
In my case since I was using custom layers anyways, I ended up just enforcing nonnegativity inside the layer definition instead.
